src/runtime/virtcontainers/vfio_endpoint.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2024 NVIDIA Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"context"
+	"fmt"
+
+	persistapi "github.com/kata-containers/kata-containers/src/runtime/virtcontainers/persist/api"
+	vcTypes "github.com/kata-containers/kata-containers/src/runtime/virtcontainers/types"
+)
+
+// VfioEndpoint represents a VFIO endpoint which claimed by guest kernel
+type VfioEndpoint struct {
+	EndpointType       EndpointType
+	HostBDF            string
+	PCIPath            vcTypes.PciPath
+	Iface              NetworkInterface
+	EndpointProperties NetworkInfo
+}
+
+// Implements Endpoint interface
+
+// Properties returns the properties of the interface.
+func (endpoint *VfioEndpoint) Properties() NetworkInfo {
+	return endpoint.EndpointProperties
+}
+
+// Name returns name of the interface.
+func (endpoint *VfioEndpoint) Name() string {
+	return endpoint.Iface.Name
+}
+
+// HardwareAddr returns the mac address of the network interface
+func (endpoint *VfioEndpoint) HardwareAddr() string {
+	return endpoint.Iface.HardAddr
+}
+
+// Type indentifies the endpoint as a vfio endpoint.
+func (endpoint *VfioEndpoint) Type() EndpointType {
+	return endpoint.EndpointType
+}
+
+// PciPath returns the PCI path of the endpoint.
+func (endpoint *VfioEndpoint) PciPath() vcTypes.PciPath {
+	return endpoint.PCIPath
+}
+
+// NetworkPair always return nil
+func (endpoint *VfioEndpoint) NetworkPair() *NetworkInterfacePair {
+	return nil
+}
+
+// SetProperties sets the properties of the endpoint.
+func (endpoint *VfioEndpoint) SetProperties(info NetworkInfo) {
+	endpoint.EndpointProperties = info
+}
+
+// SetPciPath sets the PCI path of the endpoint.
+func (endpoint *VfioEndpoint) SetPciPath(path vcTypes.PciPath) {
+	endpoint.PCIPath = path
+}
+
+// Attach for VFIO endpoint
+func (endpoint *VfioEndpoint) Attach(ctx context.Context, s *Sandbox) error {
+	return fmt.Errorf("attach is unsupported for VFIO endpoint")
```

Comment:
Why is Attach/Detach not implemented at all?

Reply:
VFIO device is attached/detached by VFIODevice's Attach()/Detach() function.